### PR TITLE
wetfish captcha implementation

### PR DESCRIPTION
**Pull request by Meleeman01**
this implements the wetfish captcha without affecting the other captchas, and can be optionally turned off for both posts and registration.

### CODE REVIEW

**Comment by Yam-s** [comment]
I implemented the captcha into registrations and posting, it should consider the different kinds of settings for those.

However, there's also a feature to require captchas for PMs which I haven't included here, but I figured someone would make a post before trying to PM someone 🤷 .

I also went ahead and fixed being unable to register because the captcha would fail.. when no captcha settings were enabled 🤦

File: forum/Themes/default/languages/Modifications.english.php [hunk 1]
```diff
@@ -34,6 +34,10 @@
 // Ban list
 $txt['ban_list_title'] = "Recent Bans";
 
+// Wetfish Captcha stuff
+$txt['wetfish_captcha'] = 'Wetfish Captcha';
+$txt['error_wetfish_captcha'] = "Please complete the captcha!";
+
```
Comment (Meleeman01) [inline forum/Themes/default/languages/Modifications.english.php:40]: that is a nice addition for sure

File: forum/Sources/Post.php [hunk 2]
```diff
@@ -1188,12 +1188,16 @@ function Post()
...
-		$context['require_verification'] = create_control_verification($verificationOptions);
-		$context['visual_verification_id'] = $verificationOptions['id'];
+		// Use SMF functionality if not using wetfish captcha.
+		if ($modSettings['visual_verification_type'] != 6)
+		{
+			require_once($sourcedir . '/Subs-Editor.php');
+			$verificationOptions = array(
+				'id' => 'post',
+			);
+			$context['require_verification'] = create_control_verification($verificationOptions);
```
Comment (Meleeman01) [inline forum/Sources/Post.php:1198]: while your fix seems to work, i've tested it every way i could think of, i think maybe exploring create_control_verification would be a good idea, that might be the root cause of why that variable was null.

**Review by Meleeman01 — COMMENTED** [review]
looks good to me

**Review by itsrachelfish — APPROVED** [review]

**Comment by itsrachelfish** [comment]
Tested on staging using two different user accounts, works great for new users and posts :+1:

**Pull request merged**